lib/helper/generateBackfillTableHelper.py

```diff
@@ -0,0 +1,138 @@
+                            # not sure what do we need to do with second value, skip handling it first
+                            current_backfill_revision_push_date_stamp = query_hg_json_obj.get("pushdate", None)[0]
+                            if current_backfill_revision_push_date_stamp:
+                                archieve_revision_relation_table[current_backfill_revision_push_date_stamp] = {"revision": current_backfill_revision,
+                                                                                                               "pkg_json_url": current_backfill_fx_pkg_json_url,
+                                                                                                               "pkg_fn_url": current_backfill_fx_pkg_fn_url,
+                                                                                                               "archive_url": current_backfill_dir_url,
+                                                                                                               "archive_dir": total_backfill_url_dict[current_backfill_dir_url]['folder_name'],
+                                                                                                               "archive_datetime": total_backfill_url_dict[current_backfill_dir_url]['folder_datetime']}
+                            else:
+                                logger.error("Cannot get corresponding datetime from pushdate key from json obj [%s]" % query_hg_json_obj)
+                        else:
+                            logger.error("Cannot get corresponding hg revision json content from url : [%s]" % query_hg_json_url)
+                    else:
+                        logger.error("Cannot get corresponding revision from moz_source_stamp key from json obj [%s]" % current_backfill_fx_pkg_json_obj)
+                else:
+                    logger.error("Cannot get corresponding fx pkg json file content from url : [%s]" % current_backfill_fx_pkg_json_url)
+            else:
+                archieve_revision_relation_table[history_backfill_url_dict[current_backfill_dir_url]["push_date_stamp"]] = {
+                    "revision": history_backfill_url_dict[current_backfill_dir_url]["revision"],
+                    "pkg_json_url": history_backfill_url_dict[current_backfill_dir_url]["pkg_json_url"],
+                    "pkg_fn_url": history_backfill_url_dict[current_backfill_dir_url]["pkg_fn_url"],
+                    "archive_url": current_backfill_dir_url,
+                    "archive_dir": history_backfill_url_dict[current_backfill_dir_url]["archive_dir"],
+                    "archive_datetime": history_backfill_url_dict[current_backfill_dir_url]["archive_datetime"]}
+
+        return archieve_revision_relation_table
+
+    @staticmethod
+    def generate_archieve_perfherder_releation_table(input_backfill_days=14, input_backfill_repo="mozilla-central", input_app_name="firefox", input_channel_name="nightly", input_white_list=[]):
```

Comment:
After having a discussion, it seems like the main entry point.
IMO, this method needs comments to tell developer `I am entry point ^.<y` and parameters and return values.